src/business_logic/transaction/objects/v2/declare_v2.rs

```diff
@@ -53,6 +55,9 @@ impl InternalDeclareV2 {
         hash_value: Option<Felt252>,
     ) -> Result<Self, TransactionError> {
         let validate_entry_point_selector = VALIDATE_DECLARE_ENTRY_POINT_SELECTOR.clone();
+        let casm_class =
```

Comment:
I think we should defer this to the execute method to only compile the sierra contract class if the validation succeeded.

Reply:
noted. changed